cpp/benchmarks/reduction/anyall_benchmark.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column_view.hpp>
+#include <cudf/reduction.hpp>
+#include <cudf/types.hpp>
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <fixture/benchmark_fixture.hpp>
+#include <synchronization/synchronization.hpp>
+
+#include <memory>
+#include <random>
+
+class Reduction : public cudf::benchmark {
+};
+
+template <typename type>
+void BM_reduction_anyall(benchmark::State& state, std::unique_ptr<cudf::aggregation> const& agg)
+{
+  const cudf::size_type column_size{(cudf::size_type)state.range(0)};
+
+  cudf::test::UniformRandomGenerator<long> rand_gen(0, 100);
```

Comment:
Potentially dumb question - can the column content affect the performance? For example, is `all` perhaps faster when most elements are zero?

Reply:
It should not make a difference. The reduction uses CUB's Reduce function which would not do an early exit since it does not really know what the `binary_op` parameter is doing.